src/tokens/slice.ts

```diff
@@ -149,19 +149,30 @@ const slice = createSlice({
       loading: false,
       error: true,
     }),
-    importToken: (state, action: PayloadAction<BaseToken>) => ({
-      ...state,
-      importedTokens: {
-        ...state.importedTokens,
-        [action.payload.tokenId]: {
-          ...action.payload,
-          // Force imported tokens to be visible even with zero balance.
-          showZeroBalance: true,
-          balance: null,
-          isManuallyImported: true,
+    importToken: (state, action: PayloadAction<StoredTokenBalance>) => {
+      const importedTokenDetails = {
+        ...action.payload,
+        // Force imported tokens to be visible even with zero balance.
+        showZeroBalance: true,
+        isManuallyImported: true,
```

Comment:
do you have plans to add this to analytics somehow separately? (or maybe it is already included by default?)

Reply:
I can add this to analytics.
In which events you think we should include this?
I'm thinking in the send flow, but maybe you had another flows in mind?

Reply:
I created a [PR](https://github.com/valora-inc/wallet/pull/4809) for this.